lib/rubygems/platform.rb

```diff
@@ -23,18 +19,40 @@ def self.local
   end
 
   def self.match(platform)
-    Gem.platforms.any? do |local_platform|
+    match_platforms?(platform, Gem.platforms)
+  end
+
+  class << self
+
+    extend Gem::Deprecate
+    rubygems_deprecate :match, "Gem::Platform.match_spec?"
```

Comment:
This might be annoying when using Bundler 1, and as long as there is the RubyGems Bundler version switcher (when removed, then Bundler 2+ would always be used and so just having a recent Bundler would avoid the issue).
Any idea how to "soft deprecate" this?

Reply:
IMO if you are on a RubyGems new enough to get this deprecation it's ok to tell you to upgrade to Bundler 2? But open to hearing from people that would be a problem for. :)

Reply:
One case I can imagine is using e.g. Ruby 2.8/3.0 or update RubyGems (which would include this code),
and then `bundle install` on a project with `Gemfile.lock` `BUNDLED WITH 1.x.y`.
Then Bundler 1 would be used (due to the version switcher) and the warning would be shown.

I think a good way to fix that is to remove the version switcher, so Bundler 2 would be used in that case.
BTW, since it seems Bundler 1 seems no longer maintained ([last release of Bundler 1](https://rubygems.org/gems/bundler/versions) in 2018), it seems important to use Bundler 2 for all cases if installed, even if the Gemfile.lock still has `BUNDLED WITH 1.x.y`.

Reply:
I filed https://github.com/rubygems/rubygems/issues/3879 about this.